plugins/nlp/nlp.cpp

```diff
@@ -68,6 +71,24 @@ namespace nlp {
         tgt = (char *) CTXMALLOC(parentCtx, tgtLen);
         memcpy_iflen(tgt, sso.str().c_str(), tgtLen);
     }
+
+    ECL_NLP_API void ECL_NLP_CALL AnalyzeTextU(unsigned & tgtLen, UChar * & tgt, size32_t anaLen, const char * ana, unsigned txtLen, UChar const * txt)
+    {
+        {
+            CriticalBlock block(cs);
+            if (nlpEng == NULL) {
+                nlpEng = new NLPEng();
+            }
+        }
+        UnicodeString const in(txt, txtLen);
+        std::string str;
+        in.toUTF8String(str);
+        StringBuffer anaBuff(anaLen,ana);
+        ostringstream sso;
+        tgtLen = nlpEng->nlpEngAnalyze(anaBuff,str.c_str(),sso);
+        tgt = (UChar *) CTXMALLOC(parentCtx, tgtLen);
+        memcpy_iflen(tgt, sso.str().c_str(), tgtLen);
```

Comment:
This looks wrong - a string (utf8?) is being copied to a UChar string - which uses double byte characters.  So the length will not be right.

There should be some test cases calling the new function (which I suspect would generate invalid output...)

Reply:
Gavin, you are right. I have been aware of a length problem but have not yet had to time sluth it out. This must be the problem.